src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -738,7 +738,67 @@ class NBDWatchCtx : public librbd::UpdateWatchCtx
   bool use_netlink;
   librados::IoCtx &io_ctx;
   librbd::Image &image;
-  unsigned long size;
+  uint64_t size;
+  std::thread notify_thread;
+  ceph::condition_variable cond;
+  ceph::mutex lock = ceph::make_mutex("NBDWatchCtx::Locker");
+  bool notify = false;
+  bool terminated = false;
+
+  bool wait_notify() {
+    dout(10) << __func__ << dendl;
+
+    std::unique_lock locker{lock};
+    cond.wait(locker, [this] { return notify || terminated; });
+
+    if (terminated) {
+      return false;
+    }
+
+    dout(10) << __func__ << ": got notify request" << dendl;
+    notify = false;
+    return true;
+  }
+
+  void notify_entry() {
```

Comment:
nit: I would prefer `handle_notify_entry` and `handle_notify_thread` names but I will don't mind if you just ignore this.